planner/core/rule_partition_processor.go

```diff
@@ -97,6 +99,12 @@ func (s *partitionProcessor) prune(ds *DataSource) (LogicalPlan, error) {
 				continue
 			}
 		}
+		// This is for `table partition (p0,p1)` syntax, only union the specified partition if has specified partitions.
+		if len(ds.partitionNames) != 0 {
```

Comment:
this `if` branch can also be removed.

Reply:
No, remove this, `select * from pattirion_table` will return 0 row.